### PR TITLE
feat: add messaging module to wrap twilio api

### LINKED ISSUE #406: Add automatic SMS and Whatsapp sending for events set up to send SMS and whatsapp to students
After event creation, if the user selected SMS and/or Whatsapp notification for students invited to an event. We need to send out SMS and Whatsapp messages to the students using the Twillio API.

Depends on: https://github.com/registreerocks/query-db-api/issues/41

 - Update the event creation schema to include fields for what contact methods should be used, as well as the message that should be sent out.
 - Add a Twillio module that supports sending SMS and Whatsapp messages to a list of numbers.
 - After event creation, send messages to all of the cellphone numbers returned by the query-db-api using the twillio module.

### PR DESCRIPTION
**Pull request by billguo99**
Resolves #406.

- [x]     Update the event creation schema to include fields for what contact methods should be used, as well as the message that should be sent out.
- [x]     Add a Twillio module that supports sending SMS ~~and Whatsapp~~ messages to a list of numbers.
- [x]     After event creation, send messages to all of the cellphone numbers returned by the query-db-api using the twillio module.

Steps done:
1. `yarn add twilio` ~~nestjs-twilio~~
2. Use nestjs cli to generate module and service for `messaging`
3. Add `twilio.config.ts` in config, to use `AppConfigService` to setup Twilio client in `MessagingModule`
4. Use ~~customdirector @InjectTwilio~~ `MESSGING_OPTIONS` to inject creds into `MessagingService` Twilio client and create simple `async sendSMS()` function

### CODE REVIEW

**Comment by billguo99** [comment]
Tested in GraphQL Playground 🎉
Mutation:
```
mutation{
  sendSMS(body: "Hi", to: "+27825893193")
}
```
Return:
```
{
  "data": {
    "sendSMS": "SM920c33ad3e8044fba8b1ff03e7f0ca90"
  }
}
```
ToDo:

- [ ] Convert/create the `sendSMS` functionality to be able to send to a list of numbers

**Comment by billguo99** [comment]
Working for list of numbers:
Mutation:
```
mutation{
  sendBulkSMS(message: "Testing duo numbers", numbers: ["+27765588901","+27825893193"])
}
```
Return:
```
{
  "data": {
    "sendBulkSMS": "SMbbef6c02daf9407aab82c03bb35277d5,SM9731072d7dd34ccdb092156e9aed0ab4"
  }
}
```

File: src/query-data/dto/query.response.ts [hunk 1]
```diff
@@ -58,4 +58,5 @@ export type DetailsDto = {
   average: number;
   race: string[];
   gender: string[];
+  smsMessage: string;
```
Comment (longtomjr) [inline src/query-data/dto/query.response.ts:61]: The query-data service will not have a smsMessage that it returns. We should remove this field.

**Review by longtomjr — CHANGES_REQUESTED** [review]
Looks good. Just a minor change

**Review by longtomjr — APPROVED** [review]
Looks good thanks!

**Pull request merged**